File: Sorts/Insertion Sort/Insertion Sort.cpp
#include <iostream>

using namespace std;



///
// Start sort function
///
void InsertionSort(int arr[], int size) {
	for (int j = 2; j < size; j++) {
		int key = arr[j];
		int i = j - 1;
		while (i >= 0 && arr[i] > key) {
			arr[i + 1] = arr[i];
			i--;
		}
		arr[i + 1] = key;
	}
}
///
// End of sort function
///


int main()
{
	///
	// Exemple of usage
	///
	int *arr;
	int size;
	cout << "n = ";
	cin >> size;

	arr = new int[size];


	///
	// Read array
	///
	for (int i = 0; i < size; i++) {
		cout << "arr[" << i << "] = ";
		cin >> arr[i];
	}



	///
	// Calling a sort function
	///
	InsertionSort(arr, size);


	///
	// Output result
	///
	for (int i = 0; i < size; i++) {
		cout << arr[i] << " ";
	}
	cout << endl;
	delete[] arr;
	return 0;
	///
	//  End of exemple of usage
	///
}

File: Sorts/Insertion Sort/Insertion Sort.py
###
## Start sort function
###
def InsertionSort(arr):
    j = 2
    while j < len(arr):
        key = arr[j]
        i = j - 1
        while i >= 0 and arr[i] > key:
            arr[i + 1] = arr[i]
            i -= 1
        arr[i + 1] = key
        j += 1


###
## End of sort function
###



###
## Exemple of usage
###

arr = []
size = int(input("n = "))

###
## Read array
###

i = 0
while i < size:
    arr.append(int(input("arr[" + str(i) + "] = ")))
    i += 1

###
## Calling a sort function
###

InsertionSort(arr)

###
## Output result
###

print(arr)

###
##  End of exemple of usage
###
